### PR TITLE
add ingressgateway into perf test run

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/istio/issues/21662
This test setup is client:1, ingressgw: 10, server:20. Do we need to modify cpu/mem resources?

### CODE REVIEW

**Comment by carolynhu** [comment]
@mandarjog FYI, here is the setup of our last customer experiments test with ingressgw: https://github.com/carolynhu/tools/commit/74a7738576e9740e8024831baa926767fab49f0b

File: perf/istio-install/values.yaml [hunk 1]
```diff
@@ -79,9 +79,9 @@ gateways:
       ver: ingress10
       app: istio-ingressgateway
       name: gateway
-    replicas: 3
-    autoscaleMin: 3
-    autoscaleMax: 5
+    replicas: 10
+    autoscaleMin: 10
```
Comment (richardwxn) [inline perf/istio-install/values.yaml]: what's the meaning if replicacount and the hpa min and max are all the same number

File: perf/benchmark/README.md [hunk 2]
```diff
@@ -199,6 +199,14 @@ python runner/runner.py --conn 1,2,4,8,16,32,64 --qps 1000 --duration 240 --perf
 This will generate corresponding `xxx_perf.data.perf` file with its `.svg` flame graph in the `perf/benchmark/flame` repo.
 Here is the [sample output](https://github.com/istio/tools/tree/master/perf/benchmark/flame/example_flame_graph/example_output)
 
+### Example 5: Run traffic through ingressgateway
+
+```base
+export INGRESSGATEWAY_IP="$(kubectl get svc -n istio-system istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+
+python runner/runner.py --headers=Host:fortioserver.local --ingress=true --conn 50,100 --qps 20000 --duration 240 --telemetry_mode=none
```
Comment (richardwxn) [inline perf/benchmark/README.md]: better to add a pointer to the values.yaml where we set the host domain

File: perf/benchmark/README.md [hunk 3]
```diff
@@ -124,8 +124,8 @@ optional arguments:
                         none, telemetryv2
   --client CLIENT       where to run the test from
   --server SERVER       pod ip of the server
-  --perf PERF           also run perf and produce flame graph
-  --ingress INGRESS     run traffic through ingress, should be a valid URL
+  --perf PERF           run perf and produce flame graph, default is False
```
Comment (richardwxn) [inline perf/benchmark/README.md]: can we rename the argument to flame_graph or sth else, the name perf is too ambiguous

File: perf/benchmark/run_benchmark_job.sh [hunk 4]
```diff
@@ -147,6 +151,8 @@ popd
 cd "${WD}/runner"
 pipenv install
 
+export SERVER_REPLICA=20
```
Comment (richardwxn) [inline perf/benchmark/run_benchmark_job.sh]: Do we need so many fortio server?

**Comment by mandarjog** [comment]
1. Ingress gateway test should use client from *outside* the cluster.
2. It does not need to use two pods, but can be used as a starting point.
3. We should be able to add multiple clients at the same time.

**Comment by jacob-delgado** [comment]
Release 1.5 branch has been closed for over a year now

**Pull request closed without merging**